libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalIndividualView/LargeCausalIndividualView/largeCausalIndividualChartUtils.ts

```diff
@@ -0,0 +1,287 @@
+  selectPointFromChartLargeData?: (data: IScatterPoint) => Promise<void>,
+  onBubbleClick?: (
+    scatterPlotData: IHighchartsConfig,
+    xSeries: number[],
+    ySeries: number[],
+    indexSeries: number[]
+  ) => void
+): Promise<IHighchartBubbleSDKClusterData | IHighchartsConfig | undefined> {
+  return await calculateBubblePlotDataFromErrorCohort(
+    cohort,
+    chartProps,
+    [],
+    jointDataset,
+    dataset,
+    isLocalCausalDataLoading,
+    true,
+    false,
+    requestBubblePlotData,
+    selectPointFromChartLargeData,
+    onBubbleClick,
+    undefined
+  );
+}
+
+export function shouldUpdateHighchart(
+  prevState: ILargeCausalIndividualChartState,
+  prevProps: ILargeCausalIndividualChartProps,
+  currentState: ILargeCausalIndividualChartState,
+  currentProps: ILargeCausalIndividualChartProps,
+  changedKeys: string[]
+): [boolean, boolean, boolean, boolean, boolean, boolean, boolean] {
+  const hasSelectedPointIndexesUpdated = !_.isEqual(
+    currentState.selectedPointsIndexes,
+    prevState.selectedPointsIndexes
+  );
+  const hasIsLocalExplanationsDataLoadingUpdated = !_.isEqual(
+    currentState.isLocalCausalDataLoading,
+    prevState.isLocalCausalDataLoading
+  );
+  const hasRevertToBubbleChartUpdated =
+    currentState.isRevertButtonClicked &&
+    prevState.isRevertButtonClicked !== currentState.isRevertButtonClicked;
+  const hasCohortUpdated = currentProps.cohort.name !== prevProps.cohort.name;
+  const hasChartPropsUpdated = !_.isEqual(
+    currentState.chartProps,
+    prevState.chartProps
+  );
+  const hasAxisTypeChanged = hasAxisTypeUpdated(
+    changedKeys,
+    prevState.chartProps,
+    currentState.chartProps
+  );
+
+  const shouldUpdate =
+    hasRevertToBubbleChartUpdated ||
+    hasSelectedPointIndexesUpdated ||
+    hasChartPropsUpdated ||
+    hasIsLocalExplanationsDataLoadingUpdated ||
+    hasCohortUpdated ||
+    hasAxisTypeChanged;
+  return [
+    shouldUpdate,
+    hasSelectedPointIndexesUpdated,
+    hasIsLocalExplanationsDataLoadingUpdated,
+    hasRevertToBubbleChartUpdated,
+    hasCohortUpdated,
+    hasChartPropsUpdated,
+    hasAxisTypeChanged
+  ];
+}
+
+export async function generateHighChartConfigOverride(
+  chartProps: IGenericChartProps | undefined,
+  hasSelectedPointIndexesUpdated: boolean,
+  hasIsLocalExplanationsDataLoadingUpdated: boolean,
+  hasRevertToBubbleChartUpdated: boolean,
+  hasChartPropsUpdated: boolean,
+  hasCohortUpdated: boolean,
+  hasAxisTypeChanged: boolean,
+  updateBubblePlotData: (chartProps: IGenericChartProps) => void,
+  updateScatterPlotData: (chartProps: IGenericChartProps) => void
+): Promise<void> {
+  if (chartProps) {
+    if (hasCohortUpdated || hasRevertToBubbleChartUpdated) {
+      updateBubblePlotData(chartProps);
+      return;
+    }
+    if (hasAxisTypeChanged) {
+      updateScatterPlotData(chartProps);
+      return;
+    }
+    if (hasChartPropsUpdated) {
+      updateBubblePlotData(chartProps);
+      return;
```

Comment:
nit: maybe use if/else and move return out to end in all of these if statements?

Reply:
I was thinking of early exit way of code. If condition satisfies take an early exit from the function. This makes sense too.

Reply:
Thanks Ilya, updated code.